external/webserver/http_client.c

```diff
@@ -871,6 +871,187 @@ void http_handle_file(struct http_client_t *client, int method, const char *url,
 	}
 }
 
+static int prepare_chunk_body(char *buf, int len, const char *body, bool is_last_msg)
+{
+	int chunk_size;
+	char httpcrnl[] = "\r\n";
+	int buflen = 0;
+
+	buflen = len;
+
+	if (is_last_msg) {
+		chunk_size = 0;
+	} else {
+		if (body == NULL) {
+			HTTP_LOGE("msg body is NULL \n");
+			return -1;
+		}
+		chunk_size = strlen(body);
+	}
+
+	//append length of message
+	buflen += snprintf(buf + buflen, HTTP_CONF_MAX_REQUEST_LENGTH - buflen, "%x", chunk_size);
+
+	//append httpcrnl
+	buflen += snprintf(buf + buflen, HTTP_CONF_MAX_REQUEST_LENGTH - buflen, "%s", httpcrnl);
+
+	if (is_last_msg) {
+		//append httpcrnl and no body
+		buflen += snprintf(buf + buflen, HTTP_CONF_MAX_REQUEST_LENGTH - buflen, "%s", httpcrnl);
+		return buflen;
+	}
+
+	if (HTTP_CONF_MAX_REQUEST_LENGTH - buflen - 2 <= chunk_size) {
+		HTTP_LOGE("msg body is larger than buffer can hold \n");
+		return -1;
+	}
+
+
+	//append body
+	buflen += snprintf(buf + buflen, HTTP_CONF_MAX_REQUEST_LENGTH - buflen, "%s", body);
+
+	//append httpcrnl
+	buflen += snprintf(buf + buflen, HTTP_CONF_MAX_REQUEST_LENGTH - buflen, "%s", httpcrnl);
+
+	return buflen;
+}
+
+static int http_send_chunk_buffer(struct http_client_t *client, char *buf)
+{
+	int  ret = 0;
+	int sndlen = 0;
+	int buflen = 0;
+
+	sndlen = strlen(buf);
+	buflen = 0;
+	while (sndlen > 0) {
+#ifdef CONFIG_NET_SECURITY_TLS
```

Comment:
Just to cross check client is checked for NULL.
Similarly chk for other required places.